rust/uefi/linux-bootloader/src/measure.rs

```diff
@@ -60,3 +67,81 @@ pub fn measure_image(system_table: &SystemTable<Boot>, image: PeInMemory) -> uef
 
     Ok(measurements)
 }
+
+/// Performs all the expected measurements for any list of
+/// companion initrds of any form.
+///
+/// Relies on the passed order of `companions` for measurements in the same PCR.
+/// You are responsible for honoring a stable order.
+pub fn measure_companion_initrds(
+    system_table: &SystemTable<Boot>,
+    companions: &[CompanionInitrd],
+) -> uefi::Result<u32> {
+    let runtime_services = system_table.runtime_services();
+    let boot_services = system_table.boot_services();
+
+    let mut measurements = 0;
+    let mut credentials_measured = 0;
+    let mut sysext_measured = false;
+
+    for initrd in companions {
```

Comment:
nit: If you filter the irrelevant initds from the vector before iterating, you can simplify this a bit. `measurements` doesn't have to be imperatively computed, but is then just `relevant_initrds.len()`. Might not be worth it, but maybe there is a way to make this function less C-like and remove the mutable state for easier reading.